From e85b662d99d6bd4bff88d8ea7eaf9a3ff82a671a Mon Sep 17 00:00:00 2001
From: Nizan Zorea <nzorea@marvell.com>
Date: Wed, 3 Aug 2016 19:38:17 +0300
Subject: [PATCH 0396/1345] dts: a7040: add device tree for NAND for
 Armada-7040-DB board

commit  24d09695e5a4e3ba858b41f23a0000cc2e528a6c from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- Add device tree for NAND setup

Change-Id: I1235410053f38315718793edf152b2f4fb56bc73
Signed-off-by: Nizan Zorea <nzorea@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31721
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../bindings/arm/marvell/armada-7k-8k.txt          |    6 +-
 arch/arm64/boot/dts/marvell/Makefile               |    2 +-
 .../arm64/boot/dts/marvell/armada-7040-db-nand.dts |  152 ++++++++++++++++++++
 3 files changed, 157 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts

diff --git a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
index 1f7ed03..8f04502 100644
--- a/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
+++ b/Documentation/devicetree/bindings/arm/marvell/armada-7k-8k.txt
@@ -23,11 +23,13 @@ Boards:
    - "marvell,armada7040-db-router", "marvell,armada7040-db"
       when the board being used is the Armada 7040 Development board
 
+   - "marvell,armada7040-db-nand", "marvell,armada7040-db"
+      when the board being used is the Armada 7040 Development board
+
    - "marvell,armada8040-db-router", "marvell,armada8040-db"
       when the board being used is the Armada 8040 Development board
 
-
 Example:
 
-compatible = "marvell,armada7040-db-router", "marvell,armada7040-db", "marvell,armada7040",
+compatible = "marvell,armada7040-db-router", "marvell,armada7040-db-nand", "marvell,armada7040-db", "marvell,armada7040",
              "marvell,armada-ap806-quad", "marvell,armada-ap806";
diff --git a/arch/arm64/boot/dts/marvell/Makefile b/arch/arm64/boot/dts/marvell/Makefile
index 36156ad..9174ff3 100644
--- a/arch/arm64/boot/dts/marvell/Makefile
+++ b/arch/arm64/boot/dts/marvell/Makefile
@@ -14,7 +14,7 @@ dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-router.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-mcbin.dtb
-
+dtb-$(CONFIG_ARCH_MVEBU) += armada-7040-db-nand.dtb
 dtb-$(CONFIG_ARCH_MVEBU) += armada-8040-db-router.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
new file mode 100644
index 0000000..a8c1f12
--- /dev/null
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2016 Marvell Technology Group Ltd.
+ *
+ * This file is dual-licensed: you can use it either under the terms
+ * of the GPLv2 or the X11 license, at your option. Note that this dual
+ * licensing only applies to this file, and not this project as a
+ * whole.
+ *
+ *  a) This library is free software; you can redistribute it and/or
+ *     modify it under the terms of the GNU General Public License as
+ *     published by the Free Software Foundation; either version 2 of the
+ *     License, or (at your option) any later version.
+ *
+ *     This library is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ * Or, alternatively,
+ *
+ *  b) Permission is hereby granted, free of charge, to any person
+ *     obtaining a copy of this software and associated documentation
+ *     files (the "Software"), to deal in the Software without
+ *     restriction, including without limitation the rights to use,
+ *     copy, modify, merge, publish, distribute, sublicense, and/or
+ *     sell copies of the Software, and to permit persons to whom the
+ *     Software is furnished to do so, subject to the following
+ *     conditions:
+ *
+ *     The above copyright notice and this permission notice shall be
+ *     included in all copies or substantial portions of the Software.
+ *
+ *     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ *     EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ *     OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *     NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ *     HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ *     WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ *     FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ *     OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/*
+ * Device Tree file for Marvell Armada 7040 development board
+ * This board file supports the NAND configuration of the board
+ */
+
+#include "armada-7040-db.dtsi"
+
+/ {
+	model = "Marvell Armada-7040 development board NAND setup";
+	compatible = "marvell,armada7040-db-nand";
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	chosen { };
+
+	ap806 {
+		config-space {
+			sdhci@6e0000 {
+				status = "okay";
+			};
+			serial@512000 {
+				status = "okay";
+			};
+			i2c@511000 {
+				status = "disabled";
+				clock-frequency = <100000>;
+			};
+		};
+	};
+
+	cpn-110-master {
+		config-space {
+			serial@702000 {
+				status = "disabled";
+			};
+			serial@702100 {
+				status = "disabled";
+			};
+			serial@702200 {
+				status = "disabled";
+			};
+			serial@702300 {
+				status = "disabled";
+			};
+			sata: sata@540000 {
+				status = "okay";
+			};
+			usb3h0: usb3@500000 {
+				status = "okay";
+			};
+			usb3h1: usb3@510000 {
+				status = "okay";
+			};
+			nand@720000 {
+				status = "okay";
+			};
+			i2c1: i2c@701000 {
+				status = "okay";
+				clock-frequency = <100000>;
+			};
+			i2c2: i2c@701100 {
+				status = "disabled";
+			};
+
+			mdio@12a200 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy2: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy3: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				emac0: mac0 {
+					phy-mode = "kr"; /* lane-2 */
+				};
+				emac2: mac2 {
+					phy-mode = "rgmii"; /* rgmii-0 */
+					phy = <&phy2>;
+				};
+				emac3: mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy3>;
+				};
+			};
+			ppv22@000000 {
+				eth0: eth0@010000 {
+					status = "okay";
+				};
+				eth1: eth1@020000 {
+					status = "okay";
+				};
+				eth2: eth2@030000 {
+					status = "okay";
+				};
+			};
+			eip197: eip197@800000 {
+				status = "okay";
+			};
+		};
+		pcie@0x600000 {
+			status = "okay";
+		};
+		pcie@0x640000 {
+			status = "okay";
+		};
+	};
+};
-- 
1.7.9.5

